### ISSUE TITLE
create: Missing parent dir for include PATH gives error (1.2 regression)

### ISSUE THREAD

**Issue by sophie-h** [borgbackup/borg#6712]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG

#### Your borg version (borg -V).

$ borg --version
borg 1.2.0

#### Operating system (distribution) and version.

Debian testing

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`borg create repo::archive /non-existent/content ~/Pictures`

## Describe the problem you're observing.

The backup aborts instead of continuing to backup the pictures.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Can easily reproduce this problem with 1.2.0 but with 1.1.7 I only get a warning, as expected.

Note: `/non-existent` works, but `/non-existent/content` does not.

This was originally reported against Pika Backup 0.4 https://gitlab.gnome.org/World/pika-backup/-/issues/220

#### Include any warning/errors/backtraces from the system logs

```
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archive.py", line 264, in OsOpen
    fd = os_open(path=path, parent_fd=parent_fd, name=name, flags=flags, noatime=noatime)
  File "/usr/lib/python3/dist-packages/borg/helpers/fs.py", line 290, in os_open
    fd = os.open(fname, _flags_noatime, dir_fd=parent_fd)
FileNotFoundError: [Errno 2] No such file or directory: '/non-existent'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5089, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5020, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 183, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 668, in do_create
    create_inner(archive, cache, fso)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 595, in create_inner
    with OsOpen(path=parent_dir, flags=flags_root, noatime=True, op='open_root') as parent_fd:
  File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/usr/lib/python3/dist-packages/borg/archive.py", line 263, in OsOpen
    with backup_io(op):
  File "/usr/lib/python3/dist-packages/borg/archive.py", line 218, in __exit__
    raise BackupOSError(self.op, exc_val) from exc_val
borg.archive.BackupOSError: open_root: [Errno 2] No such file or directory: '/non-existent'
```

**Comment by ThomasWaldmann** [comment]
your log shows that `/non-existent` does not work, but you said that this works?

**Comment by sophie-h** [comment]
> your log shows that `/non-existent` does not work, but you said that this works?

It seems to probe the parent directory. It also says `parent_dir` in the traceback.

**Comment by ThomasWaldmann** [comment]
closing as duplicate of #6374, which was fixed by #6375.
